plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,464 @@
+                    result[resource][epoch_type] = count
+        return result
+
+    def query_event_severity(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/eventSeverity".format(epoch_delta_ig_path)
+        event_severity = self.query_data(path)
+        formated_event_severity = self.format_event_severity(event_severity)
+        return formated_event_severity
+
+    def query_impacted_resource(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/impactedResource".format(epoch_delta_ig_path)
+        impacted_resource = self.query_data(path)
+        formated_impacted_resource = self.format_impacted_resource(impacted_resource)
+        return formated_impacted_resource
+
+    def format_messages(self, messages):
+        result = {}
+        for message in messages:
+            msg = message.get("message")
+            severity = message.get("severity").lower()
+            result[severity] = msg
+        return result
+
+    def query_messages(self, path):
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        if obj.get("messages") is not None:
+            result = self.format_messages(obj.get("messages"))
+            return result
+
+    def query_compliance_smart_event(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        size = 100
+        path = "{0}/smartEvents?%24epochId={1}".format(ig_base_path, compliance_epoch_id)
+        event_path = path + "&%24size={0}&%24page={1}&%24sort=-severity&category=COMPLIANCE"
+        smart_event = self.query_entry(event_path, size)
+        return smart_event
+
+    def query_msg_with_data(self, ig_name, site_name, path):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/{2}".format(ig_base_path, self.compliance_path, path)
+        result = {}
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        if obj.get("messages") is not None:
+            message = self.format_messages(obj.get("messages"))
+            if len(message) > 0:
+                result["messages"] = message
+        data = obj.get("value")["data"]
+        if len(data) > 0:
+            result["data"] = data
+        return result
+
+    def query_unhealthy_resources(self, ig_name, site_name, compliance_epoch_id):
+        result = {}
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/{1}/eventUnhealthyResources?%24epochId={2}".format(ig_base_path, self.compliance_path, compliance_epoch_id)
+        objs = self.query_data(path)
+        for obj in objs:
+            result[obj.get("bucket")] = {"count": obj.get("count"), "total": obj.get("total")}
```

Comment:
should you define defaults for the get() functions?

Reply:
Added if statement
if obj.get("bucket") and obj.get("count") and obj.get("total"):
                result[obj.get("bucket")] = {"count": obj.get("count"), "total": obj.get("total")}